be1-go/internal/state/hub_params.go

```diff
@@ -17,7 +17,7 @@ type HubParams struct {
 
 func NewHubParams(log zerolog.Logger) *HubParams {
 	return &HubParams{
-		messageChan:      make(chan socket.IncomingMessage),
+		messageChan:      make(chan socket.IncomingMessage, 100),
```

Comment:
If you want to buffer a message because the arriving order is random then you can do it with a temporary buffer that flush after x time like we did for the rumors directly inside the federation package. And you should not use buffered channel like that because it will just fill as we are handling messages sequentialy.

Reply:
And imagine that 100 messages arrive on the server while you are handling your message then you will try to send your new message on the incoming message channel but it will result to the server stalling forever.